fastplotlib/graphics/_base.py

```diff
@@ -187,23 +199,52 @@ def __eq__(self, other):
         if not isinstance(other, Graphic):
             raise TypeError("`==` operator is only valid between two Graphics")
 
-        if self.loc == other.loc:
+        if self._fpl_address == other._fpl_address:
             return True
 
         return False
 
-    def _cleanup(self):
+    def _fpl_cleanup(self):
         """
         Cleans up the graphic in preparation for __del__(), such as removing event handlers from
         plot renderer, feature event handlers, etc.
 
         Optionally implemented in subclasses
         """
-        pass
+        # clear any attached event handlers and animation functions
+        for attr in dir(self):
+            try:
+                method = getattr(self, attr)
+            except:
+                continue
+
+            if not callable(method):
+                continue
+
+            for ev_type in PYGFX_EVENTS:
+                try:
+                    self._plot_area.renderer.remove_event_handler(method, ev_type)
+                except (KeyError, TypeError):
+                    pass
+
+            try:
+                self._plot_area.remove_animation(method)
+            except KeyError:
+                pass
+
+        for child in self.world_object.children:
+            child._event_handlers.clear()
+
+        self.world_object._event_handlers.clear()
+
+        feature_names = getattr(self, "feature_events")
+        for n in feature_names:
+            fea = getattr(self, n)
+            fea.clear_event_handlers()
 
     def __del__(self):
         self.deleted = True
-        del WORLD_OBJECTS[self.loc]
+        del WORLD_OBJECTS[self._fpl_address]
```

Comment:
does this method not need to call `_fpl_cleanup`?

Reply:
oh wait, I see it getting called in subclasses

Reply:
`_fpl_cleanup()` has to be called explicitly before in order to allow `__del__()` to be called, `__del__` gets called only when the object is actually gc.